src/search/search-index-new/storage/manager.ts

```diff
@@ -23,46 +23,45 @@ export class StorageManager implements ManageableStorage {
     private _initializationResolve: Function
     private _storage: Storage
 
+    /**
+     * Handles mutation of a document to be inserted/updated to storage,
+     * depending on needed pre-processing for a given indexed field.
+     */
     private static _processIndexedField(
         fieldName: string,
         indexDef: IndexDefinition,
         fieldDef: CollectionField,
         object,
     ) {
-        if (fieldDef.fieldObject) {
-            object[fieldName] = fieldDef.fieldObject.prepareForStorage(
-                object[fieldName],
-            )
-        }
-
-        if (fieldDef._index && fieldDef.type === 'text') {
-            const fullTextField =
-                indexDef.fullTextIndexName || `_${fieldName}_terms`
-            object[fullTextField] = [...extractTerms(object[fieldName])]
+        switch (fieldDef.type) {
+            case 'text':
+                const fullTextField =
+                    indexDef.fullTextIndexName || `_${fieldName}_terms`
+                object[fullTextField] = [...extractTerms(object[fieldName])]
+                break
+            default:
         }
     }
 
     /**
      * Handles mutation of a document to be inserted/updated to storage,
-     * depending on needed processing of indexed fields.
+     * depending on needed pre-processing of fields.
      */
-    private static _processIndexedFields(def: CollectionDefinition, object) {
-        const indices = def.indices || []
+    private static _processFields(def: CollectionDefinition, object) {
+        Object.entries(def.fields).forEach(([fieldName, fieldDef]) => {
+            if (fieldDef.fieldObject) {
+                object[fieldName] = fieldDef.fieldObject.prepareForStorage(
+                    object[fieldName],
+                )
+            }
 
-        indices.forEach(indexDef => {
-            const processField = (fieldName: string) =>
+            if (fieldDef._index) {
```

Comment:
Is this ever 0 instead of null, undefined, etc.?

Reply:
Yes, very possible! Great catch! Will do a quick fix later and get this in. Thanks for the review :)